### PR TITLE
Add endpoints for renewals

### PR DESCRIPTION
**Pull request by tbille**
## Done

In every endpoint send the session so we can  authenticate the user. 

- Add endpoints:
  - Set payment method for account
```
Returns what is returned by the API: 
POST /advantage/payment-method
{
  "stripe_method_id": "<STRIPE_METHOD_ID>",
  "account_id": "<ACCOUNT_ID>"
}
```
   - Accept renewal
```
POST      /advantage/renewals/{renewal_id}/process-payment
```
   - Get renewal (returns what is returned by https://github.com/CanonicalLtd/ua-contracts/blob/develop/docs/contracts.yaml#L2187 )
```
GET     /advantage/renewals/{renewal_id}
```

Basically the flow that we need:
- setup an payment method linked to an account (via modal or whatever) -> so one POST to the first api  (`/advantage/payment-method`)
- Open a renewal that needs to be processed
- Accept the renewal (or press pay) (`/advantage/accept-renewal`)
- Those first steps could be done when approving the payment (2 calls in a row)
- Go to a view that display the status of the payment. To do that request every 5 seconds ` /advantage/renewal`

### CODE REVIEW

File: webapp/app.py [hunk 1]
```diff
@@ -112,6 +115,18 @@ def utility_processor():
 
 # Simple routes
 app.add_url_rule("/advantage", view_func=advantage_view)
+app.add_url_rule(
+    "/advantage/payment-method",
+    view_func=put_stripe_method_id,
+    methods=["PUT"],
+)
+app.add_url_rule(
+    "/advantage/renewal", view_func=get_renewal, methods=["POST"],
```
Comment (tbille) [inline webapp/app.py:124]: "/advantage/renewal/{id}", view_func=get_renewal, methods=["GET"],

File: webapp/app.py [hunk 2]
```diff
@@ -112,6 +115,18 @@ def utility_processor():
...
+)
+
+app.add_url_rule(
+    "/advantage/accept-renewal", view_func=accept_renewal, methods=["POST"],
```
Comment (tbille) [inline webapp/app.py:128]: /advantage/checkout

File: webapp/app.py [hunk 3]
```diff
@@ -112,6 +115,18 @@ def utility_processor():
 
 # Simple routes
 app.add_url_rule("/advantage", view_func=advantage_view)
+app.add_url_rule(
+    "/advantage/payment-method",
+    view_func=put_stripe_method_id,
+    methods=["PUT"],
```
Comment (tbille) [inline webapp/app.py:121]: post

File: webapp/app.py (hunk 2 above)
Comment (tbille) [reply webapp/app.py:128]: /advantage/renewals/{id}/process-payment

File: webapp/views.py [hunk 4]
```diff
@@ -225,6 +225,46 @@ def advantage_view():
...
+            return flask.jsonify({"error": "account_id required"}), 400
+
+        return advantage.put_method_id(
+            flask.session, account_id, payment_method_id
+        )
+    else:
+        return flask.jsonify({"error": "authentication required"}), 401
+
+
+def get_renewal(renewal_id=""):
```
Comment (jpmartinspt) [inline webapp/views.py:248]: `renewal_id` is always going to be set right? otherwise the url rule would not have matched?
same for the view bellow.

**Pull request merged**